### PR TITLE
Add canary workflow

### PR DESCRIPTION
**Pull request by ChristianMurphy**
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes

Follow up to: https://github.com/unifiedjs/ideas/issues/7

> Unified is a large and highly connected community of packages.
> Changes to any package could have impact downstream packages.
> Currently anticipating impact is a purely manual process.

> It could be helpful to use downstream packages, which themselves have comprehensive test suites, to help automatically detect what changes may be breaking or have unintended impacts.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -22,3 +22,45 @@ jobs:
...
+          npm run build
+      - name: checkout ${{matrix.package}}
+        uses: actions/checkout@v2
+        with:
+          repository: ${{matrix.package}}
+          path: canary/${{matrix.package}}
+      - name: test ${{matrix.package}}
+        run: |
+          npm install
+          npx rimraf "node_modules/**/unified"
```
Comment (ChristianMurphy) [inline .github/workflows/main.yml:48]: unlike micromark, this is a single package, so the folder name can be hard coded

**Review by wooorm — APPROVED** [review]

**Comment by ChristianMurphy** [comment]
unifiedjs/unified-args#27 and  rehypejs/rehype#82 have been opened to address the two build failures seen in the canary build

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -22,3 +22,45 @@ jobs:
...
+        node:
+          - node
+        package:
+          - unifiedjs/unified-engine
+          - unifiedjs/unified-args
+          - unifiedjs/unified-language-server
+          - remarkjs/remark
+          - rehypejs/rehype
+          - retextjs/retext
+          - remarkjs/react-markdown
```
Comment (ChristianMurphy) [inline .github/workflows/main.yml:66]: thoughts on including mdx in this list?

Comment (ChristianMurphy) [reply .github/workflows/main.yml:66]: added

**Comment by ChristianMurphy** [comment]
the `mdx-js/mdx` build is flagging:

```
lib/compile.js(28,43): error TS2345: Argument of type 'VFile' is not assignable to parameter of type 'VFileCompatible'.
  Type 'VFile' is not assignable to type 'VFileOptions'.
    Type 'VFile' is not assignable to type '{ [key: string]: unknown; }'.
      Index signature for type 'string' is missing in type 'VFile'.
lib/compile.js(42,47): error TS2345: Argument of type 'VFile' is not assignable to parameter of type 'VFileCompatible | undefined'.
  Type 'VFile' is not assignable to type 'VFileOptions'.
    Type 'VFile' is not assignable to type '{ [key: string]: unknown; }'.
```

https://github.com/unifiedjs/unified/runs/4669887032?check_suite_focus=true#step:6:52

which looks like it may be related to vfile/vfile/pull/64 /cc @remcohaszing 

though it is odd that https://github.com/vfile/vfile/pull/69 should (in theory) do the same test, and didn't see on `mdx-js/mdx`, it only saw it on `vfile/to-vfile` :thinking:

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -22,3 +22,46 @@ jobs:
...
+          - mdx-js/mdx
```
Comment (wooorm) [inline .github/workflows/main.yml:67]: MDX is *really* slow to build. `unified-args` and `unified-language-server` include `unified-engine`. `retext` isn’t much different from remark and rehype. I want to guard against false positives and long builds.

What about:

```
unified-engine
remark
rehype
retext
```

^-- the key direct users of unified?

Comment (ChristianMurphy) [reply .github/workflows/main.yml]: updated in 65e7911

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
sweet!

**Pull request merged**